[AIRFLOW-1186] Sort dag.get_task_instances by execution_date

OpringaoDoTurno opened the pull request:
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!


### JIRA
- [ ] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title.
    - https://issues.apache.org/jira/browse/AIRFLOW-1186


### Description
- [ ] Here are some details about my PR, including screenshots of any UI changes:
  - Adds a sort after querying task instances to avoid weird duration charts (as commented in #2226)


### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
  - There's already a (minimal) UI test for this, and this does not change existing behavior.


### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

bolkedebruin commented on airflow/www/views.py:
```diff
@@ -1421,6 +1421,7 @@ def duration(self):
 
         tis = dag.get_task_instances(
             session, start_date=min_date, end_date=base_date)
+        tis.sort(key=lambda x: x.execution_date)
```
Why not change dag.get_task_instances to return it sorted on execution date? That saves memory and is faster as it will be handled by the database

OpringaoDoTurno committed:
[AIRFLOW-1186] Sort dag.get_task_instances by execution_date

task.get_task_instances is sorted by execution_date, so we sort
dag.get_task_instances by execution_date so it doesn't break
duration chart
